app/assets/javascripts/channels/export.js

```diff
@@ -3,44 +3,55 @@ function subscribeExportChannel(uuid, url, completionCallback) {
     { channel: "ExportChannel", uuid: uuid },
     {
       connected: function() {
-        $.get(url);
+        $.ajax(url)
+          .fail(response => {
+            this.unsubscribe();
+            if (response.status === 400 && response.responseText) {
+              this.downloadError(response.responseText);
+            } else {
+              this.downloadError();
+            }
+            completionCallback();
+          });
       },
 
       // Called when the WebSocket connection is closed, either by server or by client-side stale connection monitor.
       // If the connection closes before we receive a response, count that as a failed export and unsubscribe.
       // Reconnecting is not useful because we can't tell whether the response was sent while we were disconnected.
       disconnected: function() {
-        if (newrelic) {
-          newrelic.noticeError(new Error("Export channel disconnected."), { requestedUrl: url});
-        }
-
-        const errorText = "We're sorry, your download has failed. That happens sometimes for response sets over several MB's in size.\n\n" +
-          "We're working on fixing this problem. You can monitor our progress at https://github.com/GSA/touchpoints/issues/1446. " +
-          "In the meantime, you can use our API to download large response sets. View API documentation at https://github.com/GSA/touchpoints/wiki/API.";
-
-        const blob = new Blob([errorText], {
-          type: "text/plain;charset=utf-8"
-        });
-
-        saveAs(blob, `touchpoints-failed-export-${(new Date().toISOString().slice(0, 19))}`);
+        this.downloadError('Websocket disconnected unexpectedly.');
 
         // 'this' is the subscription object
         this.unsubscribe();
         completionCallback();
       },
 
       received: function(data) {
-
-        const blob = new Blob([data.csv], {
-          type: "text/csv;charset=utf-8"
-        });
-
-        saveAs(blob, data.filename);
+        const dataUrl = data.url;
+        const filename = data.filename;
+        $.ajax(dataUrl)
```

Comment:
This call requires that the bucket be configured for CORS.

Reply:
i think we'll move to the async email soon